x/upgrade/spec/README.md

```diff
@@ -0,0 +1,20 @@
+# Upgrade Module Specification
+
+## Abstract
+
+`x/upgrade` is an implementation of a Cosmos SDK module that facilitates smoothly
+upgrading a live Cosmos chain to a new (breaking) software version. It accomplishes this by
+providing a `BeginBlocker` hook that prevents the blockchain state machine from
+proceeding once a pre-defined upgrade block time or height has been reached.
+
+The module does not prescribe anything regarding how governance decides to do an
+upgrade, but just the mechanism for coordinating the upgrade safely. Without software
+support for upgrades, upgrading a live chain is risky because all of the validators
+need to pause their state machines at exactly the same point in the process. If
+this is not done correctly, there can be state inconsistencies which are hard to
+recover from.
```

Comment:
If we do want to mention the sidecar, we could add another paragraph that says:

This module corresponds to the [cosmosd Upgradeable Binary Specification](https://github.com/regen-network/cosmosd#upgradeable-binary-specification) and cosmosd can optionally be used to fully automate the upgrade process for node operators.